### ISSUE TITLE
Expose StreamsBuilderFactoryBean thru Application context to Kafka Streams application

### ISSUE THREAD

**Issue by orchesio** [spring-cloud/spring-cloud-stream-binder-kafka#396]
I understand the idea of kafka stream binder is to hide internal boilerplate logic with Kafka Streams. However, users do need to have insight on internal metadata of Kafka Stream, especially for debugging, tuning and monitoring purposes. This could be done by exposing StreamsBuilderFactoryBean thru Application Context so that application could autowire that instance, then use its getKafkaStreams to get KafkaStream instance.

What prevent that from happening right now is that each StreamsBuilderFactoryBean is registered with an uuid, that makes the instance retrieval thru application context impossible. 

So the question is, what's the reason for appending uuid in the instance name? Can that be changed to method name as anyway the map just keep one StreamsBuilderFactoryBean for each method. With that user can easily use following code to retrieve the KafkaStreams instance.
val streamsBuilderX: StreamsBuilderFactoryBean  = context.getBean("&stream-builder-process", classOf[StreamsBuilderFactoryBean])
streamsBuilderX.getKafkaStreams.allMetadata

**Comment by sobychacko** [comment]
@orchesio Instead of using a `UUID`, we can change that to using method name. The bean name then will be `stream-builder-<method-name>`. 
Even now, if you only have a single `StreamListener` method that is using Kafka Streams, you can directly autowire the `StreamsBuilderFactoryBean`. If there are multiple such methods, then it will fail. 
We will add a fix for this.

**Comment by sobychacko** [comment]
Resolved via 38d6deb 

Each `StreamBuilderFactoryBean` is registered as `stream-builder` appended with the`StreamListener` method name. If your `StreamListener` method is named as `process` for example, the stream builder bean is named as `stream-builder-process`.
Since this is a factory bean, it should be accessed by prepending an ampersand (`&`) when accessing it programmatically. Following is an example and it assumes the `StreamListener` method is named as `process`

```
StreamsBuilderFactoryBean streamsBuilderFactoryBean = context.getBean("&stream-builder-process", StreamsBuilderFactoryBean.class);
			KafkaStreams kafkaStreams = streamsBuilderFactoryBean.getKafkaStreams();
```
